### PR TITLE
Ambient state

### PR DESCRIPTION
**Pull request by maxpkatz**
## PR summary

Add an ambient state module, and allow for physical BCs to be overwritten with ambient material.

## PR checklist

- [x] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite
- [x] all functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

File: Source/problems/bc_fill_3d.F90 [hunk 1]
```diff
@@ -37,6 +38,23 @@ subroutine hypfill(adv, adv_l1, adv_l2, adv_l3, adv_h1, adv_h2, adv_h3, &
...
+          do j = lo(2), hi(2)
+             do i = lo(1), hi(1)
+                if ((i < domlo(1) .and. (bc(1,1,1) == amrex_bc_foextrap .or. bc(1,1,1) == amrex_bc_hoextrap)) .or. &
+                    (i > domhi(1) .and. (bc(1,2,1) == amrex_bc_foextrap .or. bc(1,2,1) == amrex_bc_hoextrap)) .or. &
+                    (j < domlo(2) .and. (bc(2,1,1) == amrex_bc_foextrap .or. bc(2,1,1) == amrex_bc_hoextrap)) .or. &
+                    (j > domhi(2) .and. (bc(2,2,1) == amrex_bc_foextrap .or. bc(2,2,1) == amrex_bc_hoextrap)) .or. &
+                    (k < domlo(3) .and. (bc(3,1,1) == amrex_bc_foextrap .or. bc(3,1,1) == amrex_bc_hoextrap)) .or. &
+                    (k > domhi(3) .and. (bc(3,2,1) == amrex_bc_foextrap .or. bc(3,2,1) == amrex_bc_hoextrap))) then
+                   adv(i,j,k,URHO) = ambient_state(i,j,k,URHO)
+                end if
```
Comment (zingale) [inline Source/problems/bc_fill_3d.F90]: isn't ambient_state 1d?

Comment (maxpkatz) [reply Source/problems/bc_fill_3d.F90]: Fixed

**Comment by maxpkatz** [comment]
This PR also now modifies the wdmerger_3D test to use these new options, so there's regression suite coverage.

**Comment by zingale** [comment]
what's the status with this?

**Comment by maxpkatz** [comment]
Ready for re-review

**Pull request merged**